website/server/libs/taskManager.js

```diff
@@ -0,0 +1,170 @@
+import * as Tasks from '../models/task';
+import {
+  BadRequest,
+} from './errors';
+import Bluebird from 'bluebird';
+import _ from 'lodash';
+
+async function _validateTaskAlias (tasks, res) {
+  let tasksWithAliases = tasks.filter(task => task.alias);
+  let aliases = tasksWithAliases.map(task => task.alias);
+
+  // Compares the short names in tasks against
+  // a Set, where values cannot repeat. If the
+  // lengths are different, some name was duplicated
+  if (aliases.length !== [...new Set(aliases)].length) {
+    throw new BadRequest(res.t('taskAliasAlreadyUsed'));
+  }
+
+  await Bluebird.map(tasksWithAliases, (task) => {
+    return task.validateSync();
+  });
+}
+
+
+/**
+ * Creates tasks for a user, challenge or group.
+ *
+ * @param  req  The Express req variable
+ * @param  res  The Express res variable
+ * @param  options  Should be a user, user and challenge to create a challenge task or user and group to create a group task
```

Comment:
If we decide to use this doc style we should try to be consistent on how and where we use it